inspirehep/modules/workflows/tasks/classifier.py

```diff
@@ -38,6 +40,74 @@
 from ..utils import with_debug_logging, get_document_in_workflow
 
 
+CLASSIFIER_MAPPING = {
+    "core": "CORE",
+    "non_core": "Non-CORE",
+    "rejected": "Rejected"
+}
+
+
+def get_classifier_url():
+    """Return the classifier URL endpoint, if any."""
+    base_url = current_app.config.get('CLASSIFIER_API_URL')
+    if not base_url:
+        return None
+
+    return '{base_url}/predict/coreness'.format(base_url=base_url)
+
+
+def prepare_payload(record):
+    """Prepare payload to send to Inspire Classifier."""
+    payload = {
+        'title': get_value(record, 'titles.title[0]', ''),
+        'abstract': get_value(record, 'abstracts.value[0]', ''),
+    }
+    return payload
+
+
+@with_debug_logging
+def guess_coreness(obj, eng):
+    """Workflow task to ask inspire classifier for a coreness assessment."""
+    predictor_url = get_classifier_url()
+    if not predictor_url:
+        return None
+
+    payload = prepare_payload(obj.data)
+
+    results = json_api_request(predictor_url, payload)
+
+    if results:
```

Comment:
this is wrong as discussed IRL some time ago. You want to fail hard if the classifier doesn't work. So replace https://github.com/inspirehep/inspire-next/blob/855251f0d40436b12b3bd9c2ed5920315468f722/inspirehep/modules/workflows/utils/__init__.py#L77 with `response.raise_for_status()` and remove the `if results:` here.